vault/core.go

```diff
@@ -2093,6 +2093,49 @@ func (c *Core) emitMetrics(stopCh chan struct{}) {
 	writeTimer := time.Tick(c.counters.syncInterval)
 	identityCountTimer := time.Tick(time.Minute * 10)
 
+	// The gauge collection processes are started and stopped here
+	// because there's more than one TokenManager created during startup,
+	// but we only want one set of gauges.
+	if c.MetricSink().GaugeInterval != time.Duration(0) {
+		tokenCount1, err := c.MetricSink().NewGaugeCollectionProcess(
+			[]string{"token", "count"},
+			[]metrics.Label{{"gauge", "token_by_namespace"}},
+			func(ctx context.Context) ([]metricsutil.GaugeLabelValues, error) {
+				if c.tokenStore == nil {
+					return []metricsutil.GaugeLabelValues{}, errors.New("nil token store")
+				} else {
+					return c.tokenStore.gaugeCollector(ctx)
+				}
+			},
+			c.logger,
+		)
+		if err != nil {
+			c.logger.Error("failed to start token count", "error", err)
+		} else {
+			go tokenCount1.Run()
+			defer tokenCount1.Stop()
```

Comment:
Could we use a more descriptive variable name?

Reply:
Fixed in the new code which loops over an array instead.